packages/api/core/src/util/hook.ts

```diff
@@ -14,3 +14,18 @@ export default async (forgeConfig: ForgeConfig, hookName: string, ...hookArgs: a
   }
   await forgeConfig.pluginInterface.triggerHook(hookName, hookArgs);
 };
+
+export const runMutatingHook = async <T>(forgeConfig: ForgeConfig, hookName: string, item: T): Promise<T> => {
+  const hooks = forgeConfig.hooks;
+  if (hooks) {
+    d(`hook triggered: ${hookName}`);
+    if (typeof hooks[hookName] === 'function') {
+      d('calling mutating hook:', hookName, 'with item:', item);
+      const result = await hooks[hookName](forgeConfig, item);
```

Comment:
Why pass the forgeConfig into the hook first? This was a bit cryptic to find from the docs.